### DISCUSSION TITLE
What's the benifit of flat_hash_map over unordered_map?

### DISCUSSION THREAD

**Discussion by zzq0119** [MikePopoloski/slang#390] [Q&A] [answered]
Hi Mike, in https://sv-lang.com/commoncomp.html , the last line
```
For any hash tables that do need to live outside the stack, flat_hash_map should be used over std::unordered_map, always.
```
Could you explain why flat_hash_map is always better in such condition?

**Comment by MikePopoloski** [comment, marked as answer]
It's much much faster. std::unordered_map is notoriously slow.

**Reply by zzq0119** [reply]
OK, and the reason why "outside the stack" is emphasized here is that the time difference between the two maps would much larger than it on stack?

**Reply by MikePopoloski** [reply]
If it's only on the stack, i.e. temporary, SmallMap should be used instead to potentially avoid any heap allocation altogether. Again, just for performance reasons.

**Reply by zzq0119** [reply]
OK, thanks!
